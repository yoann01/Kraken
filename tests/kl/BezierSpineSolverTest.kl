
require Kraken;

operator entry(){
	BezierSpineSolver solver();

    Boolean debug     = true;
    Scalar rigScale   = 1.0;
    Scalar length     = 8.0;
    Mat44 base        = Xfo(Vec3(0, 1, 0)).toMat44();
    Mat44 baseHandle  = Xfo(Vec3(0, 4, 0)).toMat44();
    Mat44 tipHandle   = Xfo(Vec3(4, 6, 0)).toMat44();
    Mat44 tip         = Xfo(Vec3(4, 10, 0)).toMat44();
    Mat44 outputs[];
    outputs.resize(6);

    solver.solve(debug, rigScale, length, base, baseHandle, tipHandle, tip, outputs);

    report(outputs);
}
