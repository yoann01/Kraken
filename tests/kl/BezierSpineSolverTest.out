[FABRIC:MT] Ignored extension directory 'D:\fabric\FabricEngine-2.1.0-Windows-x86_64\Tests\Exts': unable to open directory 'D:\fabric\FabricEngine-2.1.0-Windows-x86_64\Tests\Exts': The system cannot find the path specified. (code 3 [0x3])[{row0:{x:+0.186169,y:-0.982517,z:+0.0,t:+0.0},row1:{x:+0.982517,y:+0.186169,z:+0.0,t:+1.0},row2:{x:+0.0,y:+0.0,z:+1.0,t:+0.0},row3:{x:+0.0,y:+0.0,z:+0.0,t:+1.0}},{row0:{x:+0.454748,y:-0.890619,z:+0.0,t:+0.248226},row1:{x:+0.890619,y:+0.454748,z:+0.0,t:+2.310023},row2:{x:+0.0,y:+0.0,z:+1.0,t:+0.0},row3:{x:+0.0,y:+0.0,z:+0.0,t:+1.0}},{row0:{x:+0.568249,y:-0.822856,z:+0.0,t:+0.854558},row1:{x:+0.822856,y:+0.568249,z:+0.0,t:+3.497517},row2:{x:+0.0,y:+0.0,z:+1.0,t:+0.0},row3:{x:+0.0,y:+0.0,z:+0.0,t:+1.0}},{row0:{x:+0.585514,y:-0.810662,z:+0.0,t:+1.612224},row1:{x:+0.810662,y:+0.585514,z:+0.0,t:+4.594659},row2:{x:+0.0,y:+0.0,z:+1.0,t:+0.0},row3:{x:+0.0,y:+0.0,z:+0.0,t:+1.0}},{row0:{x:+0.534319,y:-0.845282,z:+0.0,t:+2.39291},row1:{x:+0.845282,y:+0.534319,z:+0.0,t:+5.675542},row2:{x:+0.0,y:+0.0,z:+1.0,t:+0.0},row3:{x:+0.0,y:+0.0,z:+0.0,t:+1.0}},{row0:{x:+0.414627,y:-0.909991,z:+0.0,t:+3.105337},row1:{x:+0.909991,y:+0.414627,z:+0.0,t:+6.802586},row2:{x:+0.0,y:+0.0,z:+1.0,t:+0.0},row3:{x:+0.0,y:+0.0,z:+0.0,t:+1.0}}]